events/schema.py

```diff
@@ -80,11 +85,9 @@ class OccurrenceNode(DjangoObjectType):
     @classmethod
     @login_required
     def get_queryset(cls, queryset, info):
-        return (
-            queryset.annotate(enrolments_count=Count("enrolments"))
-            .select_related("event")
-            .order_by("time")
-        )
+        return queryset.annotate(
+            enrolments_count=Count("enrolments", distinct=True)
```

Comment:
Why this `distinct` is needed?

Reply:
Sometimes depends on the query, `enrolment_count` will be annotated more than one time(just graphene thing, I don't know what exact reason)  in both OccurrenceNode and the `resolve_occurrences` so distinct should be there to avoid duplicated count. It also avoids incorrect count when querying in a circle (event>occurrence>child>availableEvent>occurrence)

Reply:
Sometimes depends on the query, `enrolment_count` will be annotated more than one time(just graphene thing, I don't know what exact reason)  in both OccurrenceNode and the `resolve_occurrences` so distinct should be there to avoid duplicated count. It also avoids incorrect count when querying in a circle (event>occurrence>child>availableEvent>occurrence)

Reply:
Ok, thanks for the clarification :+1:

Reply:
What about adding a code comment that explains why you added distinct here? 

You might wonder why you did it in a few months..